src/io/iohk/scalanet/peergroup/TCPPeerGroup.scala

```diff
@@ -116,10 +116,11 @@ object TCPPeerGroup {
     override val to: InetMultiAddress = InetMultiAddress(nettyChannel.remoteAddress())
 
     override def sendMessage(message: M): Task[Unit] = {
+      val byteBuf = Unpooled.wrappedBuffer(codec.encode(message))
       toTask({
         nettyChannel
-          .writeAndFlush(Unpooled.wrappedBuffer(codec.encode(message)))
-      })
+          .writeAndFlush(byteBuf)
+      }).map(_ => byteBuf.release())
```

Comment:
If the `writeAndFlush` method throws an exception, the buffer won't be released.